llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/metrics/LlapMetricsCollector.java

```diff
@@ -101,13 +122,27 @@ void collectMetrics() {
         LlapDaemonProtocolProtos.GetDaemonMetricsResponseProto metrics =
                 client.getDaemonMetrics(null,
                         LlapDaemonProtocolProtos.GetDaemonMetricsRequestProto.newBuilder().build());
-        instanceStatisticsMap.put(identity, new LlapMetrics(metrics));
-
+        LlapMetrics newMetrics = new LlapMetrics(metrics);
```

Comment:
Why would we start the receiving of stats, if we wouldn't have a listener? Should we maybe skip this if there is no listener?

Reply:
Yes, I see several use-cases where we need statistics without the listener. For example: it might be good idea that when we chose a random node for task execution we could distribute these tasks based on the available executors